src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,386 @@
+using Octokit;
+using Org.BouncyCastle.Bcpg;
+
+namespace DependencyUpdateErrorProcessor
+{
+    /// <summary>
+    /// An instance of this class is created for each service replica by the Service Fabric runtime.
+    /// </summary>
+    internal sealed class DependencyUpdateErrorProcessor : IServiceImplementation
+    {
+        public DependencyUpdateErrorProcessor(
+            IReliableStateManager stateManager,
+            ILogger<DependencyUpdateErrorProcessor> logger,
+            BuildAssetRegistryContext context,
+            IOptions<DependencyUpdateErrorProcessorOptions> options
+            )
+        {
+            _stateManager = stateManager;
+            Logger = logger;
+            Context = context;
+            _options = options.Value;
+        }
+
+        private readonly IReliableStateManager _stateManager;
+
+        private readonly DependencyUpdateErrorProcessorOptions _options;
+
+        private ILogger<DependencyUpdateErrorProcessor> Logger { get; }
+
+        private BuildAssetRegistryContext Context { get; }
+
+        private static readonly Regex CommentMarker = new Regex(@"(\w+): '([^,\n]+)'(?:, |\))");
+
+        //Runs every hour in staging for now, in production it will run once a day.
+        [CronSchedule("0 0 0/1 1/1 * ? *", TimeZones.PST)]
+        public async Task DependencyUpdateErrorProcessing()
+        {
+            if (_options.ConfigurationRefresherEndPointUri == null && _options.DynamicConfigs == null)
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+                return;
+            }
+            await _options.ConfigurationRefresherEndPointUri.Refresh();
+            if (bool.TryParse(_options.DynamicConfigs["FeatureManagement:DependencyUpdateErrorProcessor"],
+                    out var dependencyUpdateErrorProcessorFlag) && dependencyUpdateErrorProcessorFlag)
+            {
+                IReliableDictionary<string, DateTimeOffset> update =
+                    await _stateManager.GetOrAddAsync<IReliableDictionary<string, DateTimeOffset>>("update");
+                try
+                {
+                    DateTimeOffset checkpoint;
+                    using (ITransaction tx = _stateManager.CreateTransaction())
+                    {
+                         checkpoint = await update.GetOrAddAsync(
+                         tx,
+                         "update",
+                         DateTimeOffset.UtcNow 
+                         //new DateTime(2002, 10, 18)
+                         );
+                        await tx.CommitAsync();
+                    }
+                    await CheckForErrorInRepositoryBranchHistoryTable(checkpoint, _options.GithubUrl, update);
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError(ex, "Unable to get the previous checkpoint time from reliable services.");
```

Comment:
You're not just catching getting the checkpoint here, you're catching any exception that happens inside CheckForErrorInRepositoryBranchHistoryTable as well.

Reply:
done

Reply:
This still has the same issue? You're only catching the generic exception. You don't know if this is an exception thrown from the GetOrAddAsync, or from the CheckErrorsInRepositoryBranchHistoryTableAsync method.

Reply:
is it wise to just add a general msg here?

Reply:
I added a timeout for the GetOrAddAsync method and for all the other error I created a generic error. 

We are already capturing timeout for CheckErrorsInRepositoryBranchHistoryTableAsync.